src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java

```diff
@@ -21,18 +26,34 @@ public TinyPumpkinBlock(Settings blockSettings) {
 	
 	@Override
 	public BlockState getPlacementState(ItemPlacementContext placementContext) {
-		return this.getDefaultState().with(FACING, placementContext.getPlayerFacing().getOpposite());
+		final BlockState blockState = this.getDefaultState().with(FACING, placementContext.getPlayerFacing().getOpposite());
+		if (blockState.contains(Properties.WATERLOGGED))
+		{
+			final FluidState fluidState = placementContext.getWorld().getFluidState(placementContext.getBlockPos());
+			return blockState.with(Properties.WATERLOGGED, fluidState.getFluid() == Fluids.WATER);
+		}
+		
+		return blockState;
 	}
 	
+	@Override
+	public FluidState getFluidState(BlockState blockState) {
+		return blockState.contains(Properties.WATERLOGGED) && blockState.get(Properties.WATERLOGGED) ? Fluids.WATER.getStill(false) : super.getFluidState(blockState);
+	}
 	
 	@Override
-	public VoxelShape getOutlineShape(BlockState blockState, BlockView blockView, BlockPos blockPosition, EntityContext entityContext) {
-		return Y_SHAPE;
+	public BlockState getStateForNeighborUpdate(BlockState blockState, Direction direction, BlockState blockStateOther, IWorld world, BlockPos blockPosition, BlockPos blockPositionOther) {
```

Comment:
The parameters are a bit long, some suggestions based on stuff seen in Yarn:
- The block state parameters could be just `state` and `otherState`.
- Similar thing for positions. Just `pos` and `otherPos`.